src/components/dropdown.component.js

```diff
@@ -1,43 +1,93 @@
-import $ from 'jquery';
-
-export default class DropdownComponent {
-    constructor(query) {
-        this.query = query;
-
-        this.btn = $(`${query} .dropdown-btn`);
-        this.items = $(`${query} .dropdown-item`);
-        this.list = $(`${query} .dropdown-list`);
-        this.selectedItem = $(`${query} .selected`);
-
-        this.addDropdownClickEventListner();
-        this.addItemClickEventListner();
-    }
-
-    addDropdownClickEventListner() {
-        this.btn.on('click', () => {
-            this.list.toggle();
-        });
-    }
-
-    addItemClickEventListner() {
-        this.items.on('click', (event) => {
-            this.unselectItem();
-            this.selectItem(event.currentTarget);
-            this.list.toggle();
-        });
-    }
-
-    updateDropdown() {
-        this.btn.html(this.selectedItem.text());
-    }
-
-    unselectItem() {
-        this.items.removeClass('selected');
-    }
-
-    selectItem(item) {
-        this.selectedItem = $(item);
-        this.selectedItem.addClass('selected');
-        this.updateDropdown();
-    }
-}
+import $ from 'jquery';
+
+export default class DropdownComponent {
+    constructor(query, required = true) {
+        this.query = query;
+        this.required = required;
+
+        this.container = $(`${query} .dropdown-container`);
+        this.btn = $(`${query} .dropdown-btn`);
+        this.list = $(`${query} .dropdown-list`);
+        this.items = $(`${query} .dropdown-item`);
+        this.selectedItem = $(`${query} .selected`);
+
+        this.disabled = false;
+        this.valid = false;
+
+        this.defaultText = this.btn.text();
+        this.unselectedDefaultText = this.selectedItem.text();
+
+        this.addDropdownClickEventListener();
+        this.addItemClickEventListener();
+    }
+
+    setSelection(item) {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const element of this.items) {
+            if (element.textContent === item) {
+                this.select(element);
+            }
+        }
+    }
+
+    getSelectedItem() {
+        return this.selectedItem.text();
+    }
+
+    isDisabled() {
+        return this.disabled;
+    }
+
+    isValid() {
+        return this.valid;
+    }
+
+    enable() {
+        this.container.css('pointer-events', 'auto');
+        this.container.prop('disabled', false);
+        this.disabled = this.container.is(':disabled');
+    }
+
+    disable() {
+        this.container.css('pointer-events', 'none');
+        this.container.prop('disabled', true);
+        this.disabled = this.container.is(':disabled');
+    }
+
+    validation() {

```

Comment:
Why validation doesn't return a value? If a call validation, I'll have to call another property to get the validation value?

Reply:
Now it's return this.valid.